src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -138,16 +138,33 @@ sealed class NumericRange[T](
   // based on the given value.
   private def newEmptyRange(value: T) = NumericRange(value, value, step)
 
+  // Calling length when a range contains more than Int.MaxValue causes IllegalArgumentException
+  // because the length of the range cannot be represented with Int numbers.
+  // This method tries to compare the length of this with the given Int number.
+  private def compareLength(n: Int): Int = {
+    num match {
+      case _: Numeric.ByteIsIntegral | _: Numeric.ShortIsIntegral => n compare length
+      case _ =>
+        val diff = num.minus(end, start)
```

Comment:
This doesn't work because the subtraction might overflow.  It doesn't even work for ranges that fit inside `Int`!  Consider, for instance, `NumericRange(Long.MinValue, Long.MaxValue, Int.MaxValue.toLong*4)`.

Reply:
It indeed works for ranges whose length fits inside `Int`, I also wrote the unit tests for `Byte` and `Short` and made sure about the correctness. If you notice, in the first case, if `T` is `Byte` or `Short` it'll basically use their length (`NumericRange.count`). For `Int` and `Long`, you can't even construct those with more than `Int.MaxValue` elements, at least in 2.13 which I'm using. And for `BigDecimal` and `BigInt`, the second case will work, which is what the issue actually is opened for, it also contains the appropriate tests.

Reply:
Yeah I actually came up with a better solution, I'll update the PR today, thank you!